src/ServiceControl.Monitoring/Infrastructure/EndpointRegistry.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace ServiceControl.Monitoring.Infrastructure
+{
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public class EndpointRegistry
+    {
+        ConcurrentDictionary<EndpointInstanceId, bool> endpointInstances = new ConcurrentDictionary<EndpointInstanceId, bool>();
+
+        public void Record(EndpointInstanceId endpointInstanceId)
+        {
+            endpointInstances.TryAdd(endpointInstanceId, true);    
+        }
+
+        public IDictionary<string, IEnumerable<EndpointInstanceId>> GetAllEndpoints()
+        {
+            return endpointInstances.ToArray()
+                .GroupBy(kv => kv.Key.EndpointName)
+                .ToDictionary(g => g.Key, g => g.Select(i => i.Key));
+        }
```

Comment:
Given that QueueLength is reporting EndpointInstanceIds, do we need this to return a dictionary or can it just return a list of all the instance ids?